advisor-backend/routes/schedule.js

```diff
@@ -112,12 +178,30 @@ const dfsToCount = (node, graph, visited) => {
 const findAllPrerequisites = (course, graph, prereqs) => {
   for (const prereq of graph[course]) {
     if (!prereqs.has(prereq)) {
-      prereqs.add(prereq);
+      prereqs.add(parseInt(prereq));
       findAllPrerequisites(prereq, graph, prereqs);
     }
   }
 };
 
+const findAllPostrequisites = (node, adjList, postReqSet) => {
+  for (const postreq of adjList[node]) {
+    if (!postReqSet.has(postreq)) {
+      postReqSet.add(postreq);
+      findAllPostrequisites(postreq, adjList, postReqSet);
+    }
+  }
+};
```

Comment:
This looks like it's the same as `findAllPrerequisites` except the `parseInt` - you could probably combine, but also it's a bit unexpected to me that they'd differ at all (i.e. one needing `parseInt` but the other not).